File: solutions/__015_two_sum.go
package main

// ignore 为忽略nums[ignore]这项元素
// 用于查找nums是否存在target元素
func findNum(nums []int, target, ignore int) int {
	var rtv = -1
	for i := 0; i < len(nums); i++ {
		if i == ignore {
			continue
		}
		if nums[i] == target {
			rtv = i
			break
		}
	}
	return rtv
}

func twoSum(nums []int, target int) []int {
	rtv := []int{0, 0}
	for i := 0; i < len(nums); i++ {
		rtv[1] = findNum(nums, target-nums[i], i)
		if rtv[1] != -1 {
			rtv[0] = i
			break
		}
	}
	return rtv
}

File: solutions/__015_two_sum.cpp
//
// Created by peanut on 2021/8/17.
//

#include <bits/stdc++.h>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int,int> map = {{nums[0], 0}};
    for (int i = 1; i < nums.size(); ++i) {
        auto it = map.find(target-nums[i]);
        if (it != map.end()) {
            return {it->second, i};
        }
        map[nums[i]] = i;
    }
    return {};
}
